compare.py

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/python
+#
+# Compares the speed of two different compilers on one or more benchmarks in
+# the suite.
+
+from __future__ import division, print_function
+
+import os
+import subprocess
+import sys
+import time
+
+def run_test(dir, rustc):
+    os.chdir(dir)
+
+    # First clean and rebuild from scratch. This downloads any necessary code
+    # and builds preliminary crates.
+    #
+    # We use call() instead of check_call() for `make clean` because it can
+    # fail reasonably, e.g. if we try to delete a Cargo.lock file thta isn't
+    # present.
+    subprocess.call('make clean > /dev/null 2>&1', shell=True)
+    make_env = os.environ.copy()
+    make_env['RUSTC'] = rustc
+    subprocess.check_call('make > /dev/null 2>&1', env=make_env, shell=True)
+
+    # Measure compilation speed of the crate of interest three times.
+    times = []
+    for i in range(0, 3):
+        subprocess.check_call('make touch > /dev/null 2>&1', shell=True)
+        t1 = time.time()
+        subprocess.check_call('make > /dev/null 2>&1', env=make_env, shell=True)
+        t2 = time.time()
```

Comment:
So this is interesting to me. I feel like measuring time from outside the process is pretty imprecise, though at the scales we're currently talking about this might not matter much. But also, it's just a *different* measurement than what the normal process script is doing, right? (Presumably, this is why we had to adjust all the `make touch` calls...)

I think we should land this PR as is, but I also think we should follow-up and merge the logic of the two scripts better. (For that matter, I think we should rewrite all these bash/python scripts into Rust with some common libraries, though if we just used Python that'd be fine too).

(In general scraping numbers out of the output feels like it will be more precise to me, though, and it gives finer-grained data. It'd be interesting, e.g., to be able to use this script to compare the timing for individual passes, and not just the overall run.)